terraform/README.md

```diff
@@ -1,56 +1,54 @@
 # Terraform Provider Plugin
 
-# Installation
+## Usage
 
-1. Clone the plugin:
+Please, refer to [GETTING_STARTED guide](GETTING_STARTED.md) and [official documentation](https://goteleport.com/docs/setup/guides/terraform-provider/).
 
-```bash
-git clone git@github.com:gravitational/teleport-plugins
-```
+## Development
 
-2. Install the plugin to Teleport:
+1. Install [`protobuf`](https://grpc.io/docs/protoc-installation/).
+2. Install [`protoc-gen-terraform`](https://github.com/gravitational/protoc-gen-terraform) v1.0.0+.
 
-```bash
-cd teleport-plugins/terraform
-make install
-```
+    ```go install github.com/gravitational/protoc-gen-terraform@a63aa54956b6bbbcdac039d2f54261bae12d19e8```
 
-3. Configure teleport:
+    _NOTE_: Once PR is merged, we'll replace SHA with v1.0.0
 
-```bash
-tctl create example/teleport.yaml
-tctl auth sign --format=file --user=terraform --out=terraform-identity --ttl=10h
-```
+3. Install [`Terraform`](https://learn.hashicorp.com/tutorials/terraform/install-cli) v1.1.0+. Alternatively, you can use [`tfenv`](https://github.com/tfutils/tfenv). Please note that on Mac M1 you need to specify `TFENV_ARCH` (ex: `TFENV_ARCH=arm64 tfenv install 1.1.6`).
 
-Move generated keys to the desired location.
+4. Clone the plugin:
 
-4. If you desire to use an example for testing:
+    ```bash
+    git clone git@github.com:gravitational/teleport-plugins --branch chore/terraform-refactoring
+    ```
 
-```bash
-cp example/vars.tfvars.example example/vars.tfvars
-```
+    _NOTE_: Once PR is merged, we'll remove --branch
 
-Edit `vars.tfvars` and set path to certificate files which were generated in the previous step.
+5. Build and install the plugin:
 
-# Regenerating the schema
+    ```bash
+    cd teleport-plugins/terraform
+    make install
+    ```
 
-```
-go install github.com/gravitational/protoc-gen-terraform
-make gen-schema
-```
+6. Run tests:
 
-Please note that `ProvisionTokenV2.Allow` field is defined lowercase in `.proto` file (`repeated TokenRule allow = 2 [ (gogoproto.jsontag) = "allow,omitempty" ];`). You need to manually patch `types_tfschema.go`, set `Name: "Allow"` in provision token metadata struct (`GenSchemaMetaProvisionTokenV2()`) until it is fixed.
+    ```bash
+    make test
```

Comment:
That's weird. Testing framework supposed to be self sustained: it downloads required binaries and should work without license key. I'll try to do tests from scratch tomorrow. I possibly forgot to add feature flag for enterprise entities. `TESTING.md` exists in the root folder (testing framework is shared between plugins) - however, it might be a bit outdated, I'll take a look.